### ISSUE TITLE
(@aws-cdk/aws-lambda-python): BundlingOptions for PythonLayerVersion are ignored

### ISSUE THREAD

**Issue by cponfick-bhs** [aws/aws-cdk#19522]
### What is the problem?

It appears that asset_hash_type and asset_hash are ignored in the bundling parameter of the PythonLayerVersion. The asset_hash_type used is always AssetHashType.SOURCE. This is a problem if the requirement file includes a env variable which changes on every deploy, e.g a token.

### Reproduction Steps

```python

# token = required from aws cli (changes every time)
# lock_file_hash = hash from Pipfile.lock (does not changes)
# entry = path to the directory containing the Pipfile.lock

self.base_python_layer = PythonLayerVersion(self, "BasePythonLayer",
                                                    entry=entry,
                                                    description="Dependencies from Pipfile.lock.",
                                                    compatible_runtimes=[Runtime.PYTHON_3_8],
                                                    bundling=BundlingOptions(
                                                        environment={"PYPI_MIRROR_TOKEN": token},
                                                        asset_hash_type=AssetHashType.CUSTOM,
                                                        asset_hash=lock_file_hash
                                                    ), compatible_architectures=[Architecture.X86_64])
```
The token is the only thing, which is changing in the entry repository. On every cdk deploy call.

### What did you expect to happen?

Update the layer only when the `lock_file_hash` variable changes.

### What actually happened?

The layer is updated every time. 

### CDK CLI Version

2.17.0

### Node.js Version

v17.7.2

### OS

macOS Monterey 12.3

### Language

Python

### Language Version

Python (3.8.9)

### Other information

This issue is somewhat related to #19508, since I tried to create a custom hash as a answer to the missing exclude possibility.

In my opinion it is also a security issue uploading the requirements file to the layer, since it might contain user and password information to private pypi mirrors. In our case a token with a short expiration time was the solution, but it might not be for other users.

**Comment by corymhall** [comment]
@cponfick-bhs can you try to reproduce this using a very simple hash (e.g. `asset_hash="abc"`). I've
not been able to reproduce this issue and that should help us narrow down the issue.

**Comment by cponfick-bhs** [comment]
> @cponfick-bhs can you try to reproduce this using a very simple hash (e.g. `asset_hash="abc"`). I've not been able to reproduce this issue and that should help us narrow down the issue.

@corymhall I am able to reproduce it by using `asset_hash="1"`. The layer is updated as long as files in the `entry` change. Hence, I assume `asset_hash_type=AssetHashType.Source` is still used despite changing it to Custom.

In my case the changing file is the requirements.txt file, containing the auth token. When I use the same token in sequential deploys. The layer is not updated.

**Comment by corymhall** [comment]
ah ok it looks like it's because the hash that is calculated includes the `BundlingOptions` which is
changing every time for you. I think the solution to this should be #19508 so you don't have to try
to hack around it.

**Comment by cponfick-bhs** [comment]
@corymhall do I understand this correctly. When using a custom hash, this hash is used to create another hash based on the `BundlingOptions`? This does not sound right to me.

**Comment by corymhall** [comment]
Yes that is correct. We create a new hash using the custom hash and bundling options as inputs. This is because the bundling options instruct CDK _how_ the asset should be bundled, so if that changes then the hash should be invalidated. Imagine you passed a different Docker image or an environment variable that would affect the output of bundling.

https://github.com/aws/aws-cdk/blob/f1c9c4e311ba4075092c647cf4062d0705517dc4/packages/%40aws-cdk/core/lib/asset-staging.ts#L487-L518
